frontend/src/app/modules/home/dashboard/dashboard.component.ts

```diff
@@ -21,33 +25,61 @@ export class DashboardComponent implements OnInit, OnDestroy {
         private router: Router,
         private dashboardService: DashboardService,
         private updateDataService: ShareDataService<Dashboard>,
-        private deleteDataService: ShareDataService<number>
-    ) { }
+        private deleteDataService: ShareDataService<number>,
+        private toastNotificationService: ToastNotificationService,
+        private spinnerService: SpinnerService
+    ) {
+        super();
+    }
 
     ngOnInit(): void {
-        this.updateSubscription$ = this.updateDataService.currentMessage
-            .subscribe((dashboard) => {
-                this.dashboard = dashboard;
-            });
-
+        this.spinnerService.show();
         this.route.params.subscribe(params => {
             const { id } = params;
             this.showTileMenu = false;
-            this.dashboard = this.dashboardService.get(id);
+            this.dashboardService.get(id)
+                .pipe(this.untilThis)
+                .subscribe(resp => {
+                    this.dashboard = resp.body;
```

Comment:
It's better to request it as not a FullRequest, then you don't need to read from the body. Or you can map body in service. It's not a good practice to work with HTTP context in components.